huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go

```diff
@@ -0,0 +1,536 @@
+package workspace
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/workspace/v2/services"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+type AuthenticationType string
+
+const (
+	// The authentication type of Workspace service.
+	AuthTypeLocal    AuthenticationType = "LITE_AS"
+	AuthTypeAdDomain AuthenticationType = "LOCAL_AD"
+)
+
+func adDomainSchemaResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"admin_account": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"active_domain_ip": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"active_domain_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"standby_domain_ip": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"standby_domain_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"active_dns_ip": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"standby_dns_ip": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"delete_computer_object": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func securityGroupSchemaResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func ResourceService() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServiceCreate,
+		ReadContext:   resourceServiceRead,
+		UpdateContext: resourceServiceUpdate,
+		DeleteContext: resourceServiceDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
```

Comment:
suggest to add an attribute named `status`

Reply:
done.